components/brave_wallet/browser/internal/hd_key.cc

```diff
@@ -69,17 +72,35 @@ const secp256k1_context* GetSecp256k1Ctx() {
   return kSecp256k1Ctx;
 }
 
+std::vector<uint8_t> GeneratePublicKey(Secp256k1PrivateKeySpan private_key) {
+  secp256k1_pubkey public_key;
+  if (!secp256k1_ec_pubkey_create(GetSecp256k1Ctx(), &public_key,
+                                  private_key.data())) {
+    LOG(ERROR) << "secp256k1_ec_pubkey_create failed";
```

Comment:
Most logging should be removed before merge and at the very least this should be VLOG instead of error logs. Also using a log in place of returning actual error information using base::expected is an anti-pattern.

Reply:
I agree we don't need this logging. This is just some moved code which I didn't want to change much